openml/_api_calls.py

```diff
@@ -324,6 +324,24 @@ def _send_request(request_method, url, data, files=None, md5_checksum=None):
                     raise NotImplementedError()
                 __check_response(response=response, url=url, file_elements=files)
                 if request_method == "get" and not __is_checksum_equal(response.text, md5_checksum):
+
+                    # -- Check if encoding is not UTF-8 perhaps
+                    # Get checksum for binary response content
+                    md5 = hashlib.md5()
+                    md5.update(response.content)
+                    md5_checksum_content = md5.hexdigest()
+
+                    # Check if checksum is equal to the expected checksum and raise better error
+                    if md5_checksum_content == md5_checksum:
```

Comment:
```suggestion 
                    if __is_checksum_equal(response.content, md5_checksum):
```

Reply:
This won't work as `__is_checksum_equal()` assumes UTF-8 encoding and text as input. `response.content` is already a binary. I could rewrite `__is_checksum_equal()` to allow both if you prefer that?

Reply:
Ah my bad, I did a quick patch locally yesterday (trying to read the files to convert to parquet), and forgot I actually also changed `__is_checksum_equal` itself. I think it would be cleaner to rewrite `__is_checksum_equal`. I'm impartial to wether that means the function taking both str and bytes, or if you accept only bytes and call it with `repsonse.text.encode('utf-8')` first.

Reply:
I updated it accordingly